### PR TITLE
fix: fix some paste bugs

### LINKED ISSUE #1998: [Bug] Copy the number format in excel and paste it into univer without retaining the format
### 在您提交此问题之前，您是否检查了以下内容？

- [X] 这真的是个问题吗？
- [X] 我已经在 Github Issues 中搜索过了，但没有找到类似的问题。

### 受影响的包和版本

dev

### 复现链接

1、复制表格中的格式粘贴到univer
[样张.xlsx](https://github.com/dream-num/univer/files/15036154/default.xlsx)

### 预期行为

保留数字格式

### 实际行为

显示为数值
image

### 运行环境

_No response_

### 系统信息

_No response_

### PR DESCRIPTION
**Pull request by ybzky**
close #1998 https://github.com/dream-num/univer-pro/issues/1323 https://github.com/dream-num/univer-pro/issues/1322 https://github.com/dream-num/univer-pro/issues/1342

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/sheets-ui/src/controllers/clipboard/utils.ts [hunk 1]
```diff
@@ -354,6 +355,15 @@ export function getSetCellStyleMutations(
         if (withRichFormat && value.p?.body) {
             newValue.p = value.p;
         }
+        const content = String(value.v);
+        const numfmtValue = numfmt.parseDate(content) || numfmt.parseTime(content) || numfmt.parseNumber(content);
```
Comment (Gggpound) [inline packages/sheets-ui/src/controllers/clipboard/utils.ts:359]: 这里的number能处理哪些情况？会不会漏？

File: packages/sheets-ui/src/controllers/clipboard/utils.ts [hunk 2]
```diff
@@ -354,6 +355,15 @@ export function getSetCellStyleMutations(
...
+        if (numfmtValue?.z && newValue.s && typeof newValue?.s === 'object') {
```
Comment (Gggpound) [inline packages/sheets-ui/src/controllers/clipboard/utils.ts:360]: 这里的s如果不存在呢？就不处理了吗

**Review by Gggpound — APPROVED** [review]

File: packages/sheets-ui/src/controllers/clipboard/utils.ts (hunk 2 above)
Comment (ybzky) [reply packages/sheets-ui/src/controllers/clipboard/utils.ts:360]: 有道理 我改下

**Comment by zhaolixin7** [comment]
@ybzky 复制外部内容粘贴到univer后格式未被覆盖

**Pull request merged**